cassandra-mesos-model/src/main/proto/io/mesosphere/mesos/frameworks/cassandra/model.proto

```diff
@@ -18,15 +18,75 @@ option java_outer_classname = "CassandraFrameworkProtos";
 message CassandraFrameworkConfiguration {
     optional string frameworkId = 1;
     optional string frameworkName = 2;
-    optional string cassandraVersion = 3;
-    optional int32 numberOfNodes = 4;
-    optional double cpuCores = 5;
-    optional int64 memMb = 6;
-    optional int64 diskMb = 7;
     optional int64 healthCheckIntervalSeconds = 8;
-    optional int32 numberOfSeeds = 9;
     optional int64 bootstrapGraceTimeSeconds = 10;
     repeated PortMapping portMapping = 11;
+
+    // default configuration
+    required CassandraConfigRole defaultConfigRole = 6;
+    // additional configurations (currently unused)
+    repeated CassandraConfigRole configRoles = 7;
+}
+
+// CassandraConfigRole is currently only used once. But different nodes may require different sizings -
+// e.g. when replacing old metal with new metal. Different DCs may also have different sizings.
+// Individual nodes just reference a "config role".
+message CassandraConfigRole {
+    // arbitrary name of this configuration - usually "default" but could be something like "16core-128g-4x500gSSD"
+    optional string name = 1;
+
+    optional string cassandraVersion = 2;
+    optional int32 numberOfNodes = 3;
+    optional double cpuCores = 4;
+    optional int64 diskMb = 6;
+    optional int32 numberOfSeeds = 7;
+
+
+    // total memory required (must be greater than memJavaHeapMb + memAssumeOffHeapMb)
+    optional int64 memMb = 5;
+
+    // Cassandra memory usage can be categorized into
+    // 1. Java heap (MAX_HEAP_SIZE) - including new-gen (HEAP_NEWSIZE)
+    //    amount of Java heap in MB
+    //    (defaults to 50% of memMb, if not present)
+    optional int64 memJavaHeapMb = 8;
+    // 2. Off-Heap (roughly 50% of Java heap)
+    //    amount of Off heap in MB
+    //    (defaults to 50% of memMb, if not present)
+    optional int64 memAssumeOffHeapMb = 9;
```

Comment:
Right. The default off-heap requirement is about 50% of the Java heap. So using 50% of the provisioned `memMb` should work. That's still too few IMO since we really want to have room left for block cache.